pkg/channel/event_dispatcher.go

```diff
@@ -71,25 +73,25 @@ func NewEventDispatcher(logger *zap.Logger) *EventDispatcher {
 // DispatchEvent dispatches an event to a destination over HTTP.
 //
 // The destination and reply are URLs.
-func (d *EventDispatcher) DispatchEvent(ctx context.Context, event cloudevents.Event, destination, reply string) error {
+func (d *EventDispatcher) DispatchEvent(ctx context.Context, event cloudevents.Event, destination, reply *apis.URL) error {
 	var err error
 	// Default to replying with the original event. If there is a destination, then replace it
 	// with the response from the call to the destination instead.
 	response := &event
-	if destination != "" {
+	if destination != nil {
 		destinationURL := d.resolveURL(destination)
 		ctx, response, err = d.executeRequest(ctx, destinationURL, event)
 		if err != nil {
 			return fmt.Errorf("unable to complete request to %s: %v", destinationURL, err)
 		}
 	}
 
-	if reply == "" && response != nil {
+	if reply == nil && response != nil {
```

Comment:
Looking at the logic here, I wonder if an early-return on:

```golang
if response == nil {
  return nil
}
```

Would make the remaining logic simpler. e.g.

```golang

if response == nil {...}
if reply == nil {...}
...
```

Reply:
done